### DISCUSSION TITLE
set과 list 차이에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by 3Juhwan** [2024-woowacourse-study/level-interview#92] [Java]
.

**Comment by alstn113** [comment]
Set과 List는 모두 자바 컬렉션 프레임워크의 인터페이스입니다.

차이점은 다음과 같습니다.

1. 중복 허용 여부

   - List: 중복된 요소를 허용합니다. 즉, 리스트에 동일한 요소를 여러 번 추가할 수 있습니다.
   - Set: 중복된 요소를 허용하지 않습니다. 즉, 집합에는 동일한 요소가 두 번 이상 포함되지 않습니다.

2. 요소의 순서

   - List: 요소의 순서가 유지됩니다. 요소가 리스트에 추가된 순서대로 저장되며, 인덱스를 사용하여 요소에 직접 접근할 수 있습니다.
   - Set: 요소의 순서가 보장되지 않습니다. Set은 요소를 순서 없이 저장하며, 요소의 순서는 구현체에 따라 다를 수 있습니다.

3. 중복 확인 방법:

   - List: 요소의 추가 시에 중복을 확인하지 않습니다. 따라서 중복된 요소를 추가할 수 있습니다.
   - Set: 요소의 추가 시에 중복을 확인하여 이미 존재하는 요소인 경우 추가를 거부합니다. hashCode 메서드를 호출해서 같은 해시 코드가 있는지를 확인합니다.

4. 주요 구현체

   - List: ArrayList, LinkedList 등이 List 인터페이스를 구현한 대표적인 클래스입니다.
   - Set: HashSet, TreeSet 등이 Set 인터페이스를 구현한 대표적인 클래스입니다.

요소의 중복 여부와 순서에 따라 데이터를 저장하고 관리해야 하는 경우 List를 사용하며, 중복을 허용하지 않고 순서가 중요하지 않은 경우 Set을 사용합니다.

**Comment by GIVEN53** [comment]
||List|Set|
|---|:---:|:---:|
|중복 허용|O|X|
|순서 보장|O|X|

List: 데이터의 입력 순서가 중요한 경우나 동일한 항목을 여러 번 포함해야 하는 경우 사용한다.
Set: 순서가 중요하지 않고 중복을 허용하지 않는 데이터를 저장할 때, 또는 집합 연산을 수행해야 하는 경우 사용한다.
